xgi/algorithms/simpliciality.py

```diff
@@ -0,0 +1,434 @@
+from itertools import chain, combinations
+
+import numpy as np
+from scipy.special import binom
+
+from ..core import Hypergraph
+from ..utils import Trie
+
+__all__ = [
+    "edit_simpliciality",
+    "simplicial_edit_distance",
+    "face_edit_simpliciality",
+    "mean_face_edit_distance",
+    "simplicial_fraction",
+]
+
+
+def edit_simpliciality(H, min_size=2, exclude_min_size=True):
+    """Computes the edit simpliciality.
+
+    The number of edges needed to be added
+    to a hypergraph to make it a simplicial complex.
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        The hypergraph of interest
+    min_size: int, default: 2
+        The minimum hyperedge size to include when
+        calculating whether a hyperedge is a simplex
+        by counting subfaces.
+    exclude_min_size : bool, optional
+        Whether to include minimal simplices when counting simplices, by default True
+
+    Returns
+    -------
+    float
+        The edit simpliciality
+
+    References
+    ----------
+    "The simpliciality of higher-order order networks"
+    by Nicholas Landry, Jean-Gabriel Young, and Nicole Eikmeier,
+    *EPJ Data Science* **13**, 17 (2024).
+    """
+    edges = H.edges.filterby("size", min_size, "geq").members()
+
+    t = Trie()
+    t.build_trie(edges)
+
+    maxH = Hypergraph(
+        H.edges.maximal()
+        .filterby("size", min_size + exclude_min_size, "geq")
+        .members(dtype=dict)
+    )
+    if not maxH.edges:
+        return np.nan
+
+    ms = 0
+    for id1, e in maxH.edges.members(dtype=dict).items():
+        redundant_missing_faces = set()
+        for id2 in maxH.edges.neighbors(id1):
+            if id2 < id1:
+                c = maxH._edge[id2].intersection(e)
+                if len(c) >= min_size:
+                    redundant_missing_faces.update(_missing_subfaces(t, c, min_size))
+
+                    # we don't have to worry about the intersection being a max face
+                    # because a) there are no multiedges and b) these are all maximal
+                    # faces so no inclusions.
+                    if not t.search(c):
+                        redundant_missing_faces.add(frozenset(c))
+
+        nm = _max_number_of_subfaces(min_size, len(e))
+        nf = _count_subfaces(t, e, min_size)
+        rmf = len(redundant_missing_faces)
+        ms += nm - nf - rmf
+
+    try:
```

Comment:
Do we have a style preference for a try/catch block here rather than checking for 0 denominator in a normal conditional? Since `s` and `ms` should both be integers, could we just check `if (ms+s) > 0` rather than relying on exception handling?

Reply:
I think this comment applies to the returns of all of these functions. Practically the outcome should be the same using either approach, so in my view it is a style/readability question. My personal preference is for a conditional, but open to leaving it as is if it is consistent with style elsewhere or desireable for another reason.

Reply:
Great point. I fixed this according to your suggestion.